src/_guides/language/effective-dart/usage.md

```diff
@@ -1047,7 +1047,7 @@ Future<int> countActivePlayers(String teamName) [!async!] {
 {% prettify dart %}
 Future<int> countActivePlayers(String teamName) {
   return downloadTeam(teamName).then((team) {
-    if (team == null) return 0;
```

Comment:
Wow, I assumed that this would be auto-wrapped.

Reply:
It will at runtime, but type inference gets a little confused when you have a function body that returns both an immediate value and a future, I think.